src/engraving/libmscore/engravingobject.h

```diff
@@ -274,6 +274,9 @@ class EngravingObject
     void writeProperty(XmlWriter& xml, Pid id) const;
     void writeStyledProperties(XmlWriter&) const;
 
+    virtual void added() {}
+    virtual void removed() {}
```

Comment:
What do you think about renaming these methods to onAdded and onRemoved? Also, I would suggest moving these methods under the protected section

Reply:
well, I prefer to not use such names anywhere except of qml to be honest

Reply:
onAdded and onRemoved indicate the thing being the event handler itself, not the notifier (cfr javascript/qml)